pkg/apis/monitoring/v1alpha1/scrapeconfig_types.go

```diff
@@ -101,6 +101,11 @@ type ScrapeConfigSpec struct {
 	// Authorization header to use on every scrape request.
 	// +optional
 	Authorization *v1.SafeAuthorization `json:"authorization,omitempty"`
+	// HTTP scheme to use for scraping.
+	// If empty, Prometheus uses the default value `http`.
+	// +kubebuilder:validation:Enum=http;https
```

Comment:
https://github.com/prometheus-operator/prometheus-operator/pull/5675/files#r1230477468

Reply:
again, this is where the Prometheus config format and the Kube conventions differ. While Prometheus expects `http` or `https`, Kubernetes [recommends](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#constants) `HTTP` and `HTTPS` for constant values:

> When an acronym or initialism each letter in the acronym should be uppercase, such as with ClientIP or TCPDelay

The role of the operator is to convert (lowercase) the constant values when generating the config.